scripts/parse-speakers.js

```diff
@@ -0,0 +1,138 @@
+
+'use strict'
+
+const fs = require('fs')
+const path = require('path')
+const es = require('event-stream')
+const slug = require('slug')
+
+// Read the "cleaned" version of the speaker readme
+fs.createReadStream('./scripts/speakers-clean.md')
+  // Split at empty lines - empty lines separate the records
+  .pipe(es.split(/^$/m))
+  // Create an array from the text which contains the speaker's info
+  .pipe(es.map(function(data, callback) {
+    const lines = data.split('\n')
+    const len = lines.length
+    const speakerInfo = lines.splice(1, len-2)
+      .map((l) => l.trim())
+      .filter((l) => l.length > 0 ? true : false)
+    callback(null, speakerInfo)
+  }))
+  // Convert the speaker's info to an object
+  .pipe(es.through(function write(data) {
+    const speaker = parseSpeaker(data)
+    this.emit('data', speaker)
+  }))
+  // Build the speaker's hugo markdown data
+  .pipe(es.map(function(data, callback) {
+    const obj = buildSpeaker(data)
+    callback(null, JSON.stringify(obj))
+  }))
+  // Write the speaker's markdown content to /content/speaker
+  .pipe(es.map(function(data, callback) {
+    writeSpeakerFile(JSON.parse(data), () => {
+      callback(null, data)
+    })
+  }))
+  // Profit.
+  .on('end', () => console.log('\nDone!\n'))
+
+
+/**
+ * @description Generates a speaker's markdown for consumption by hugo
+ */
+function buildSpeaker(data) {
+  const company = data.company || ''
+  const obj = {
+    data
+  }
+  let buff = ['+++',
+    'bio = ""',
+    `company = "${company}"`,
+    `date = "${new Date().toISOString()}"`,
+    'location = ""',
+    `name = "${data.name}"`,
+    'subjects = []',
+    `title = "${data.title}"\n`
+  ]
+
+  data.links.forEach((link, index) => {
+    let urlStr = `  url = "${link.href}"`
+
+    if (index !== data.links.length - 1) {
+      urlStr += '\n'
+    }
+
+    buff = buff.concat([
+      '[[social_links]]',
+      `  name = "${link.text}"`,
+      urlStr
+    ])
+
+  })
+
+  buff.push('+++\n')
+  obj.markdown = buff.join('\n')
+  return obj
+
+}
+
+/**
+ * @description Writes a speaker's info to their own markdown file
+ */
+function writeSpeakerFile(speaker, callback) {
+  const filename = slug(`${speaker.data.name}-${speaker.data.title}`)
```

Comment:
For the sake of pretty URLs, how do you feel about the slug being `slug(`${speaker.data.name})`? And then if the file already exists, append `-1` onto it or `-2`, etc.